### PR TITLE
feat: Add support for logging via DataDog

### PR DESCRIPTION
**Pull request by chimedpalden**
## Description
This PR Adds the Datadog logger. I'll be removing all other logger once the Datadog integration is tested successfully on preview and on production.

https://github.com/jalantechnologies/boilerplate-mern/issues/87

Screenshot 2023-12-18 at 12 14 54 PM

### CODE REVIEW

File: src/apps/backend/modules/logger/internals/datadog-logger.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import * as winston from 'winston';
+
+import ConfigService from '../../config/config-service';
+
+import WinstonLogger from './winston-logger';
+
+export default class DatedogLogger extends WinstonLogger {
```
Comment (bbbneo333) [inline src/apps/backend/modules/logger/internals/datadog-logger.ts:7]: typo - `DatedogLogger`

File: src/apps/backend/modules/logger/internals/loggers.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import ConfigService from '../../config/config-service';
 import { UnknownTransportError } from '../types';
 
 import ConsoleLogger from './console-logger';
+import DatedogLogger from './datadog-logger';
```
Comment (bbbneo333) [inline src/apps/backend/modules/logger/internals/loggers.ts:5]: typo - `DatedogLogger`

**Review by bbbneo333 — CHANGES_REQUESTED** [review]
Please fix typos.

Also, from ticket, is this requirement being fulfilled?

```
A log should have clear details on environment (staging, production, etc..)
```

If yes - Can post example of a screenshot where it has the environment?

**Comment by chimedpalden** [comment]
> Please fix typos.
> 
> Also, from ticket, is this requirement being fulfilled?
> 
> ```
> A log should have clear details on environment (staging, production, etc..)
> ```
> 
> If yes - Can post example of a screenshot where it has the environment?

@bbbneo333 fixed the typo. The environment can be determined by the service name for eg: boilerplate-mern-preview or boilerplate-mern-production.

Screenshot 2023-12-18 at 4 17 59 PM

File: src/apps/backend/modules/logger/internals/types.ts [hunk 3]
```diff
@@ -15,4 +15,5 @@ export enum LoggerTransport {
   Rollbar = 'rollbar',
   Grafana = 'grafana',
   Papertrail = 'papertrail',
+  Datadog = 'datadog',
```
Comment (jjalan) [inline src/apps/backend/modules/logger/internals/types.ts:18]: Should we remove support for all other logger types such as Rollbar, Grafana, Papertrail etc to reflect one opiniated approach on monitoring / logging from boilerplate standpoint or would that be part of separate PR?

Comment (chimedpalden) [reply src/apps/backend/modules/logger/internals/types.ts:18]: yes, @jjalan i'll remove it in a separate PR. I just wanted to test the end-to-end flow from logging to alert in the MS team via datadog before removing other options.

**Review by bbbneo333 — APPROVED** [review]

**Comment by bbbneo333** [comment]
LGTM!
@chimedpalden - Please make sure appropriate env vars are already set in doppler before merging the PR.
